src/tv2-common/parts/effekt.ts

```diff
@@ -172,44 +192,55 @@ export function CreateEffektForPartInner<
 	}
 }
 
-export function CreateMixForPartInner(
+export function CreateMixEffectBlueprintPieceForPart(
 	pieces: IBlueprintPiece[],
 	externalId: string,
 	durationInFrames: number,
-	layers: {
-		sourceLayer: string
-		casparLayer: string
-		sisyfosLayer: string
-	}
+	sourceLayer: string
 ): Pick<IBlueprintPart, 'inTransition'> {
+	const tags = [
+		GetTagForTransition(
+			literal<ActionTakeWithTransitionVariantMix>({
+				type: 'mix',
+				frames: durationInFrames
+			})
+		)
+	]
+	const effectName: string = 'mix'
+	addEffectBlueprintPiece(pieces, durationInFrames, externalId, effectName, sourceLayer, tags)
+	return createInTransitionForEffect(durationInFrames)
+}
+
+function addEffectBlueprintPiece(
+	pieces: IBlueprintPiece[],
+	durationInFrames: number,
+	externalId: string,
+	name: string,
+	sourceLayer: string,
+	tags: string[]
+): void {
 	pieces.push(
 		literal<IBlueprintPiece>({
 			enable: {
 				start: 0,
 				duration: Math.max(TimeFromFrames(durationInFrames), 1000)
 			},
 			externalId,
-			name: `MIX ${durationInFrames}`,
-			sourceLayerId: layers.sourceLayer,
+			name: `${name.toUpperCase()} ${durationInFrames}`,
+			sourceLayerId: sourceLayer,
 			outputLayerId: SharedOutputLayers.JINGLE,
 			lifespan: PieceLifespan.WithinPart,
-			tags: [
-				GetTagForTransition(
-					literal<ActionTakeWithTransitionVariantMix>({
-						type: 'mix',
-						frames: durationInFrames
-					})
-				)
-			],
+			tags,
 			content: {
 				timelineObjects: [],
 				ignoreMediaObjectStatus: true
 			}
 		})
 	)
+}
 
+function createInTransitionForEffect(durationInFrames: number): Pick<IBlueprintPart, 'inTransition'> {
```

Comment:
Is Effect used as `EFFECT 100` or as a category (mix, dip, cut, effekt) of all the transition types?

Reply:
"Effect" is used as the name for the category. The actual name is `MixEffects`, but since there is an effect called `Mix` I was hesitant to call the method `createInTransitionForMixEffect` as that would most likely lead to more confusion.

Reply:
`createInTransition` is too broad. All our BlueprintParts can have an InTransition and they will have different behaviour. In this case the `partContentDelayDuration` is hardcoded to zero.

Also, I had to export the function doing some other refactoring which makes it dangerous to just have it called `CreateInTransition` since you would be tempted to always use it for whatever you need an InTransition for.
It would be a completely different scenario if we instead of just exported all of our functions (which is a horrible practice) created classe instead that had the functions, then we could use your naming suggestion because it would then read `effekt.createInTransition(...)` when we called it

I have refactored it to be `CreateInTransitionForAtemTransitionStyle`.